pkg/shell/topnav.jsx

```diff
@@ -123,11 +123,15 @@
 
         if (!this.superuser_connection || (this.superuser_connection.options.host !=
                                            this.props.machine.connection_string)) {
-            if (this.superuser_connection)
+            if (this.superuser_connection) {
                 this.superuser_connection.close();
+                this.superuser_connection = null;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18991-20230629-070021-013c8682-fedora-38-devel-storage//Coverage/lcov/github-pr.diff.gcov.html)